### PR TITLE
Rework index initialisation logic

### PR DESCRIPTION
**Pull request by rsdy**
This PR's aim is to make the qdrant dependency mandatory. In practice, we always want semantic indexing, paying attention to this every time we use this subsystem is not super productive.

In addition, steps are taken to reduce some of the dependency mess between tantivy & semantic indexes. `FileCache` is now hierarchically adjacent to the `File` indexer, which opened the door to removing a bunch of database dependnencies for that layer. The `FileCache` is now managed by `SyncHandle`, and passed on as a reference.

The initialization flow is also somewhat clearer, and at a high level is contained fully in `lib.rs` for easier reading.

### CODE REVIEW

File: server/bleep/src/cache.rs [hunk 1]
```diff
@@ -390,20 +393,18 @@ impl<'a> FileCache<'a> {
         &self,
         cache_keys: &CacheKeys,
         repo_name: &str,
-        repo_ref: &str,
+        repo_ref: &RepoRef,
```
Comment (ggordonhall) [inline server/bleep/src/cache.rs:396]: Can update the comment on L390

File: server/bleep/src/indexes/file.rs [hunk 2]
```diff
@@ -42,10 +39,11 @@ use crate::{
 struct Workload<'a> {
     cache: &'a FileCacheSnapshot<'a>,
     file_filter: &'a FileFilter,
+    repo_ref: &'a RepoRef,
     repo_disk_path: &'a Path,
     repo_name: &'a str,
     repo_metadata: &'a RepoMetadata,
-    repo_ref: String,
+    repo_ref_str: String,
```
Comment (ggordonhall) [inline server/bleep/src/indexes/file.rs:46]: Confusing having both `repo_ref` and `repo_ref_str` here. Why don't we just have `repo_ref` and transform into a `String` when needed.

**Review by ggordonhall — APPROVED** [review]
Nice! The initialisation logic is now much easier to follow along.

File: server/bleep/src/indexes/file.rs [hunk 3]
```diff
@@ -117,9 +114,10 @@ impl Indexable for File {
 
                 let workload = Workload {
                     repo_disk_path: &repo.disk_path,
-                    repo_ref: reporef.to_string(),
+                    repo_ref_str: reporef.to_string(),
                     repo_name: &repo_name,
                     file_filter: &file_filter,
+                    repo_ref: reporef,
```
Comment (nerdypepper) [inline server/bleep/src/indexes/file.rs:120]: do we need these both? `workload.repo_ref_str` is deriavable from `workload.repo_ref.to_string()` no?

**Review by nerdypepper — APPROVED** [review]

**Pull request merged**